source/slang/ir.cpp

```diff
@@ -3889,103 +3867,65 @@ namespace Slang
             return false;
         }
 
-        IROp opA = IROp(a->op & kIROpMeta_PseudoOpMask);
-        IROp opB = IROp(b->op & kIROpMeta_PseudoOpMask);
+        const IROp opA = IROp(a->op & kIROpMeta_PseudoOpMask);
+        const IROp opB = IROp(b->op & kIROpMeta_PseudoOpMask);
 
         if (opA != opB)
         {
             return false;
         }
 
+        // Both are types
+        if (IRType::isaImpl(opA))
+        {
+            if (IRBasicType::isaImpl(opA))
+            {
+                // If it's a basic type, then their op being the same means we are done
+                return true;
+            }
+
+            // We don't care about the parent or positioning
+            // We also don't care about 'type' - because these instructions are defining the type.
+            // 
+            // We may want to care about decorations.
+
+            // If it's a resource type - special case the handling of the resource flavor 
+            if (IRResourceTypeBase::isaImpl(opA) &&
+                static_cast<const IRResourceTypeBase*>(a)->getFlavor() != static_cast<const IRResourceTypeBase*>(b)->getFlavor())
+            {
+                return false;
+            }
+
+            // TODO(JS): There is a question here about what to do about decorations.
+            // For now we ignore decorations. Are two types potentially different if there decorations different?
+            // If decorations play a part in difference in types - the order of decorations presumably is not important.
+
+            // All the operands of the types must be equal
+            return _areTypeOperandsEqual(a, b);
```

Comment:
I think I may have confused things in my previous review, because I think you still need `_hasNominalEquality` (or `_isNominalType`) here, to be able to write:

```
if(_hasNominalEquality(a))
    return false; // They weren't pointer-identical, so they can't be equal
else
    return _areTypeOperandsEqual(a,b); // are they structurally equal?
```

Reply:
To be clear about the issue: with the code as in this PR I think that any two `IRStructType`s would be considered equal. They both have `kIROp_StructType`, and they both have equal operands (since structure types have zero operands; they just have *children*).